cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
```

Comment:
No need to add `_H` here and in all other `.h` file, the file later can be renamed to `.hpp`, there is no need to restrict ourselves to `.h` forever.